src/rqt_graph/rosgraph2_impl.py

```diff
@@ -300,21 +301,43 @@ def _graph_refresh(self):
         subscriptions = defaultdict(list)
         servers = defaultdict(list)
 
+        publisher_topic_names = set()
+        subscriber_topic_names = set()
+
         for name, namespace in self._node.get_node_names_and_namespaces():
             node_name = namespace + name if namespace.endswith('/') else namespace + '/' + name
 
             for topic_name, topic_type in \
                     self._node.get_publisher_names_and_types_by_node(name, namespace):
                 publishers[topic_name].append(node_name)
+                publisher_topic_names.add(topic_name)
 
             for topic_name, topic_type in \
                     self._node.get_subscriber_names_and_types_by_node(name, namespace):
                 subscriptions[topic_name].append(node_name)
+                subscriber_topic_names.add(topic_name)
 
             for service_name, service_type in \
                     self._node.get_service_names_and_types_by_node(name, namespace):
                 servers[service_name].append(node_name)
 
+        publisher_qos = {}
```

Comment:
If there are two publishers in the graph with different qos, only one will be shown.
It's true that showing all the qos will be a bit verbose for a tool like rqt, but in some cases showing only one may be confusing (e.g. one of the publishers is matching the subscribers in the topic, the other not because of qos incompatibility).

Maybe, in that case is better to show a message: "publishers with {N} different QoS were found".

Reply:
> If there are two publishers in the graph with different qos, only one will be shown.

Not in the graph - but in a single node publishing to the same topic. Then yes, only one would be shown which is indeed not nice.

Showing both (or all) shouldn't be a problem. Until now parallel edges were consolidated. Please see 331dc12644d08ba8cf6ca704f9579ef3a14e8eb3 which collects all QoS and then uses parallel edged for different QoS. E.g. if there are 3 publishers with 2 different QoS the graph will show 2 parallel edges each showing the unique QoS.